### PR TITLE
rgw:make bucket created in slave zonegroup can sync its acl .

### PR DESCRIPTION
**Pull request by snakeAngel2015**
Signed-off-by: snakeAngel2015 <li.jieA@h3c.com>

### CODE REVIEW

**Comment by snakeAngel2015** [comment]
@yehudasa @oritwas

**Comment by oritwas** [comment]
@snakeAngel2015 , if you are fixing a bug can you open an issue in tracker.ceph.com.
and add:
Fixes: <url to the tracker issue>
in the git commit message

**Comment by snakeAngel2015** [comment]
@oritwas
Ok,I have added it ,plz check it ,thanks
http://tracker.ceph.com/issues/16888

**Comment by oritwas** [comment]
@snakeAngel2015 
it should be Fixes: not Fix: in the commit message

**Comment by snakeAngel2015** [comment]
@oritwas 
I have corrected it .Plz check it ,thanks
Fixex:http://tracker.ceph.com/issues/16888

**Comment by snakeAngel2015** [comment]
I have corrected it ,Plz checked it ,thanks .
Fixes:http://tracker.ceph.com/issues/16888

**Comment by snakeAngel2015** [comment]
@oritwas
I have corrected it ,Plz checked it ,thanks .
Fixes:http://tracker.ceph.com/issues/16888

**Comment by dachary** [comment]
jenkins test this please (no logs)

File: src/rgw/rgw_common.cc [hunk 1]
```diff
@@ -166,6 +166,8 @@ void req_info::rebuild_from(req_info& src)
   effective_uri.clear();
   host = src.host;
 
+  env->set("CONTENT_TYPE", src.env->get("CONTENT_TYPE"));
```
Comment (oritwas) [inline src/rgw/rgw_common.cc]: What if there is no Content-type header?
(some admin command don't have content type header)
Please check before setting it.

**Comment by snakeAngel2015** [comment]
@oritwas 
 I have modified it ,plz checked it ,thanks

**Comment by yehudasa** [comment]
@snakeAngel2015 we need a ceph tracker ticket for this one, and reference it in the commit message. Also, we need to make sure we test this use case in our multisite suite (@cbodley)

**Comment by snakeAngel2015** [comment]
@yehudasa
I have already added it  days ago,Plz checked it ,thanks .
 Fixes:http://tracker.ceph.com/issues/16888

File: src/rgw/rgw_op.cc [hunk 2]
```diff
@@ -4071,6 +4071,14 @@ void RGWPutACLs::execute()
   if (op_ret < 0)
     return;
 
+  if (!store->is_meta_master() && (!store->get_zonegroup().equals(s->bucket_info.zonegroup))) {
+     op_ret = forward_request_to_master(s, NULL, store, in_data, NULL);
+     if (op_ret < 0) {
+       ldout(s->cct, 20) << __func__ << "forward_request_to_master returned ret=" << op_ret << dendl;
+     }
+    return;
```
Comment (oritwas) [inline src/rgw/rgw_op.cc]: you only need to return here forward request failed

**Comment by oritwas** [comment]
fixed in #14082

**Comment by oritwas** [comment]
fixed in #14082

**Pull request closed without merging**